Lib/sre_compile.py

```diff
@@ -268,7 +280,7 @@ def _optimize_charset(charset, fixup, fixes):
                     else:
                         charmap[av] = 1
                 elif op is RANGE:
-                    r = range(av[0], av[1]+1)
+                    r = r0 = range(av[0], av[1]+1)
```

Comment:
At line 296.
```
if fixup and not hascased:
    hascased = any(map(iscased, r0))
```
This intentionally moved after the loops that set bits in `charmap` because for large ranges like `range(0x20000, 0x110000)` it is better to get an IndexError earlier (the size of `charmap` is 0x100 or 0x10000) than burn CPU in long useless loop.